joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/LambdaTests.scala

```diff
@@ -1,130 +1,667 @@
+                     |  }
+                     |}
+                     |""".stripMargin)
+
+    // Only test the method node for type info, since this is effectively the same test as above
+    // with a different source for the expected type.
+    "create a method node for the lambda" in {
+      cpg.typeDecl.name("Foo").method.name(".*lambda.*").l match {
+        case List(lambdaMethod) =>
+          // Lambda body creation tested separately
+          lambdaMethod.name shouldBe "lambda$0"
+          lambdaMethod.fullName shouldBe "Foo.lambda$0:java.lang.String(java.lang.String)"
+          lambdaMethod.parameter.l match {
+            case List(lambdaInput) =>
+              lambdaInput.name shouldBe "lambdaInput"
+              lambdaInput.typeFullName shouldBe "java.lang.String"
+
+            case result => fail(s"Expected single lambda parameter but got $result")
+          }
+          lambdaMethod.methodReturn.typeFullName shouldBe "java.lang.String"
+
+        case result => fail(s"Expected single lambda method but got $result")
+      }
+    }
+  }
+
+  "lambdas returned from a function" should {
+    val cpg = code("""
+        |import java.util.function.Function;
+        |
+        |public class Foo {
+        |  public Function<String, String> test(String input, String fallback) {
+        |    return lambdaInput -> lambdaInput.length() > 5 ? "Long" : fallback;
+        |  }
+        |}
+        |""".stripMargin)
+
+    // Only test the method node for type info, since this is effectively the same test as above
+    // with a different source for the expected type.
+    "create a method node for the lambda" in {
+      cpg.typeDecl.name("Foo").method.name(".*lambda.*").l match {
+        case List(lambdaMethod) =>
+          // Lambda body creation tested separately
+          lambdaMethod.name shouldBe "lambda$0"
+          lambdaMethod.fullName shouldBe "Foo.lambda$0:java.lang.String(java.lang.String)"
+          lambdaMethod.parameter.l match {
+            case List(lambdaInput) =>
+              lambdaInput.name shouldBe "lambdaInput"
+              lambdaInput.typeFullName shouldBe "java.lang.String"
+
+            case result => fail(s"Expected single lambda parameter but got $result")
+          }
+          lambdaMethod.methodReturn.typeFullName shouldBe "java.lang.String"
+
+        case result => fail(s"Expected single lambda method but got $result")
+      }
+    }
+  }
+
+  "lambdas capturing instance vars" should {
+    val cpg = code("""
+        |import java.util.function.Consumer;
+        |
+        |public class Foo {
+        |    public String s;
+        |
+        |    public static void sink(String s) {}
+        |
+        |    public Consumer<String> test() {
+        |        return input -> sink(input + s);
+        |    }
+        |}
+        |""".stripMargin)
+
+    // TODO: Add 0th parameter logic
+    "create a 0th `this` parameter" ignore {
```

Comment:
why the ignore?

Reply:
Because if I didn't ignore it, then it would fail ;) Fixing the issue isn't a bad idea though